openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,559 @@
+	}
+	if a.nodeListSynced == nil {
+		return fmt.Errorf("%s plugin needs a node lister synced", PluginName)
+	}
+	return nil
+}
+
+func (a *managementCPUsOverride) Admit(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) error {
+	if attr.GetResource().GroupResource() != coreapi.Resource("pods") || attr.GetSubresource() != "" {
+		return nil
+	}
+
+	pod, ok := attr.GetObject().(*coreapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attr, fmt.Errorf("unexpected object: %#v", attr.GetObject()))
+	}
+
+	podAnnotations := map[string]string{}
+	for k, v := range pod.Annotations {
+		podAnnotations[k] = v
+	}
+
+	// strip any resource annotations specified by a user
+	stripResourcesAnnotations(pod.Annotations)
+	// strip any workload annotation to prevent from underlying components(CRI-O, kubelet) to apply any changes
+	// according to the workload annotation
+	stripWorkloadAnnotations(pod.Annotations)
+
+	workloadType, err := getWorkloadType(podAnnotations)
+	if err != nil {
+		invalidError := getPodInvalidWorkloadAnnotationError(podAnnotations, err.Error())
+		return errors.NewInvalid(coreapi.Kind("Pod"), pod.Name, field.ErrorList{invalidError})
+	}
+
+	// no workload annotation is specified under the pod
+	if len(workloadType) == 0 {
+		return nil
+	}
+
+	workloadAnnotation := fmt.Sprintf("%s%s", podWorkloadTargetAnnotationPrefix, workloadType)
+	effect, err := getWorkloadAnnotationEffect(podAnnotations[workloadAnnotation])
+	if err != nil {
+		invalidError := getPodInvalidWorkloadAnnotationError(podAnnotations, fmt.Sprintf("failed to get workload annotation effect: %v", err))
+		return errors.NewInvalid(coreapi.Kind("Pod"), pod.Name, field.ErrorList{invalidError})
+	}
+
+	// TODO: currently we support only PreferredDuringScheduling effect
+	if effect != workloadEffectPreferredDuringScheduling {
+		invalidError := getPodInvalidWorkloadAnnotationError(podAnnotations, fmt.Sprintf("only %q effect is supported", workloadEffectPreferredDuringScheduling))
+		return errors.NewInvalid(coreapi.Kind("Pod"), pod.Name, field.ErrorList{invalidError})
+	}
+
+	if !a.waitForSyncedStore(time.After(timeToWaitForCacheSync)) {
+		return admission.NewForbidden(attr, fmt.Errorf("%s node and namespace caches not synchronized", PluginName))
+	}
+
+	// allow annotations on project to override management pods CPUs requests
+	ns, err := a.getPodNamespace(attr)
+	if err != nil {
+		return err
+	}
+
+	if !doesNamespaceAllowWorkloadType(ns.Annotations, workloadType) {
+		return admission.NewForbidden(attr, fmt.Errorf("%s the pod namespace %q does not allow the workload type %s", PluginName, ns.Name, workloadType))
+	}
+
+	nodes, err := a.nodeLister.List(labels.Everything())
+	if err != nil {
+		return admission.NewForbidden(attr, err) // can happen due to informer latency
+	}
+
+	// it possible to run static pods without available nodes under the cluster, but the static pod containers created
```

Comment:
static pods should be skipped at the very beginning before any mutation takes place.  Admission is not a voting member of what static pods are running.  Any mutation is simply deceptive since all the decisions are made in advance of the pod being created.

Reply:
See my comment above.

Reply:
done